core/include/kernel/tee_misc.h

```diff
@@ -83,6 +83,15 @@ bool _core_is_buffer_intersect(vaddr_t b, size_t bl, vaddr_t a, size_t al);
 
 /*  UUID string: "XXXXXXXX-XXXX-XXXX-XXXXXXXXXXXXXXXX". Size includes '\0' */
 #define TEE_UUID_STRING_LEN		36
-int uuid2str(char *dst, TEE_UUID *uuid);
+int uuid2str(char *dst, const TEE_UUID *uuid);
+
+static inline char *_uuid2str(const TEE_UUID *uuid)
+{
+	static char str[TEE_UUID_STRING_LEN];
```

Comment:
Yeah, I know. I'm not really happy with that either.
What are the alternatives?
1. Use buffers allocated on the stack? A bit ugly IMO. Especially since DMSG() is a nop in release mode, so the buffers would need to be guarded with `#ifdef`'s or qualified with `__unused`... Huh.
2. Define macros such as:
```
#define UUID_FMT "%08x-%04x-%04x-%02x%02x%02x%02x%02x%02x%02x%02x"
#define UUID_VALUES(_u) (_u).timeLow, (_u).timeMid, (_u).timeHiAndVersion, /* and so on */
```
...then do `DMSG("Blah " UUID_FMT, UUID_VALUES(uuid));`. Not very nice either, but maybe a good compromise.
3. Introduce a new format specifier to DMSG() etc. (i.e., to `kprintf()` in `lib/libutils/isoc/snprintf.c` basically). The Linux kernel as `%pU` for this. From a usage standpoint I think it is the best option, but the code in `kprintf()` is a bit tricky so that is not the first thing I considered ;)
- Any other idea?